.circleci/config.yml

```diff
@@ -0,0 +1,633 @@
+    keys:
+      - *m2_cache_key
+      - pom.xml2-
+  m2_repo_path: &m2_repo_path
+             /home/circleci/.m2/repository
+  pepper_apis_path: &pepper_apis_path
+            /home/circleci/repo/pepper-apis
+  circleci_path: &circleci_path
+            /home/circleci/repo/.circleci
+  builds_bucket: &builds_bucket
+            ddp-build-artifacts
+  jar_build_bucket_dir: &jar_build_bucket_dir
+        /ddp-study-server-pepper-apis
+  housekeeping_test_db: &housekeeping_test_db
+      housekeepingdb
+  studyserver_test_db: &studyserver_test_db
+      studyservicedb
+  test_parallellism: &test_parallelism 15
+  ddp_server_port: &ddp_server_port 5556
+
+
+executors:
+  build-executor:
+    docker:
+      - image: broadinstitute/study-server-build:java-2020-04-15A
+  test-executor:
+    docker:
+      - image: broadinstitute/study-server-build:java-2020-04-20A
+      # start pubsub emulator
+      - image: broadinstitute/study-server-build:pubsub-1
+      - image: circleci/mysql:5.7-ram
+        environment: &DB_VARS
+          MYSQL_ROOT_PASSWORD: rootpw
+          MYSQL_DATABASE: test_db
+          MYSQL_USER: user
+          MYSQL_PASSWORD: passw0rd
+    resource_class: medium+ #default medium not big enough for tests
+
+  deploy-executor:
+    docker: # todo: create new image endpoint, add this in admin tab for faster starts
+    - image: *ci_support_image
+
+commands:
+  build-jar:
+    description: Build jar
+    parameters:
+      pom_file:
+        type: string
+      m2_repo:
+        type: string
+        default: *m2_repo_path
+    steps:
+      - restore_cache: *m2_cache_keys
+      - run:
+          name: Run maven package
+          command: |
+            mvn package -DskipTests -file << parameters.pom_file >> -Dcheckstyle.skip -Dmaven.repo.local=<< parameters.m2_repo >>
+  setup-shared-env:
+    description: "Setup shared ENV vars used by multiple scripts"
+    steps:
+      - run:
+          name: Set environment to be used by build
+          command: |
+            echo "export VAULT_TOKEN=$(vault write -field=token auth/approle/login role_id=$VAULT_ROLE_ID secret_id=$VAULT_ROLE_SECRET_ID)" >> $BASH_ENV
+            echo "export SHORT_GIT_SHA=$(echo $CIRCLE_SHA1 | cut -c 1-7)" >> $BASH_ENV
+            source $BASH_ENV
+
+  set-deployment-env:
+    description: "Determine the deploy target and set the ENVIRONMENT var"
+    steps:
+      - run:
+          name: Determine deployment environment for branch << pipeline.git.branch >>
+          command: |
+            case "${CIRCLE_BRANCH}" in
+            develop)
+              DEPLOY_ENV=dev
+            ;;
+            # @todo remove when done
+              study-server-parallel-tests)
+              DEPLOY_ENV=dev
+            ;;
+            test)
+              DEPLOY_ENV=test
+            ;;
+            test_deleteme1)
+              DEPLOY_ENV=dev
+            ;;
+            staging)
+              DEPLOY_ENV=staging
+            ;;
+            prod)
```

Comment:
I think this deploy branch should be named `production`.

Reply:
66b170b

Reply:
Wait... can you swap this? The branch should be `production` and the env should be `prod`.

Reply:
Note to self: when making changes at the end of the day be triply careful as you are more likely to make stupid mistakes.
ff3dd42c